turbo/snapshotsync/freezeblocks/caplin_snapshots.go

```diff
@@ -280,21 +282,65 @@ Loop:
 		s.segmentsMax.Store(segmentsMax)
 	}
 	s.segmentsReady.Store(true)
+
+	s.calculateVisibleSegments()
 	s.idxMax.Store(s.idxAvailability())
 	s.indicesReady.Store(true)
 
 	return nil
 }
 
-func (s *CaplinSnapshots) idxAvailability() uint64 {
-	var beaconBlocks uint64
-	for _, seg := range s.BeaconBlocks.segments {
-		if seg.Index() == nil {
+func (s *CaplinSnapshots) calculateVisibleSegments() {
+	var maxIndex = -1
+	var maxBlockNum uint64
+	if len(s.BeaconBlocks.dirtySegments) == 0 {
+		return
+	}
+	slices.SortFunc(s.BeaconBlocks.dirtySegments, func(i, j *Segment) int {
+		return cmp.Compare(i.from, j.from)
+	})
+
+	if s.BeaconBlocks.maxVisibleBlock.Load() != 0 &&
```

Comment:
atomic.Load - must be used once - because next Load call may return another result 

also: Result of this func is: maxVisibleBlock.Store(maxBlockNum)
Unclear why need read this property in the beginning of func

Reply:
Updated to used only once.

Here's going to append new visible segments to existing visible segments. 
I want to check if there's gap between them.